### PR TITLE
[tests-only] [full-ci] Added Then steps to improve checks in tests for apiShareManagementBasicToRoot suite

### PR DESCRIPTION
**Pull request by grgprarup**
## Description
This PR adds `Then` steps to existing API test scenarios to better check the results of `When` steps.

**Suites covered:**
- `apiShareManagementBasicToRoot`

## Related Issue

- Part of https://github.com/owncloud/core/issues/39512

## How Has This Been Tested?

- Locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Review by SagarGi — COMMENTED** [review]
lgtm :+1:

**Review by SwikritiT — COMMENTED** [review]
LGTM

File: tests/acceptance/features/apiShareManagementBasicToRoot/createShareToRootFolder.feature [hunk 1]
```diff
@@ -400,12 +402,14 @@ Feature: sharing
     And user "Alice" has uploaded file with content "some content" to "lorem.txt"
     When user "Alice" shares file "lorem.txt" with group "grp1" using the sharing API
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiShareManagementBasicToRoot/createShareToRootFolder.feature:403]: can be used as given

File: tests/acceptance/features/apiShareManagementBasicToRoot/createShareToRootFolder.feature [hunk 2]
```diff
@@ -400,12 +402,14 @@ Feature: sharing
...
     And the administrator adds user "Carol" to group "grp1" using the provisioning API
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiShareManagementBasicToRoot/createShareToRootFolder.feature:404]: should be the only when step.

**Review by amrita-shrestha — COMMENTED** [review]
https://github.com/owncloud/core/pull/39953/files#diff-8385a5dbd0742c1fdc75be3b05d3b23795ad8579b86d67f0e4294523e05166cfR481 can shift to **Given** step

**Review by amrita-shrestha — COMMENTED** [review]
https://github.com/owncloud/core/pull/39953/files#diff-8385a5dbd0742c1fdc75be3b05d3b23795ad8579b86d67f0e4294523e05166cfR495
https://github.com/owncloud/core/pull/39953/files#diff-8385a5dbd0742c1fdc75be3b05d3b23795ad8579b86d67f0e4294523e05166cfR512
can shift to **Given** step

**Review by amrita-shrestha — COMMENTED** [review]
https://github.com/owncloud/core/pull/39953/files#diff-8385a5dbd0742c1fdc75be3b05d3b23795ad8579b86d67f0e4294523e05166cfR379-R380
Can remove two-line we check HTTP and OCS status codes of when step only

File: tests/acceptance/features/apiShareManagementBasicToRoot/createShareToRootFolder.feature [hunk 3]
```diff
@@ -398,14 +398,16 @@ Feature: sharing
       | grp1      |
     And user "Brian" has been added to group "grp1"
     And user "Alice" has uploaded file with content "some content" to "lorem.txt"
-    When user "Alice" shares file "lorem.txt" with group "grp1" using the sharing API
-    And the administrator adds user "Carol" to group "grp1" using the provisioning API
-    Then the content of file "lorem.txt" for user "Brian" should be "some content"
+    And user "Alice" has shared file "lorem.txt" with group "grp1"
+    When the administrator adds user "Carol" to group "grp1" using the provisioning API
+    Then the OCS status code of responses on all endpoints should be "<ocs_status_code>"
+    And the HTTP status code of responses on all endpoints should be "200"
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiShareManagementBasicToRoot/createShareToRootFolder.feature:404]:
```suggestion
    Then the OCS status code should be "<ocs_status_code>"
    And the HTTP status code should be "200"
```

File: tests/acceptance/features/apiShareManagementBasicToRoot/createShareToRootFolder.feature [hunk 4]
```diff
@@ -473,8 +476,8 @@ Feature: sharing
     And user "Alice" has created folder "userZeroFolder"
     And user "Alice" has shared folder "userZeroFolder" with user "Brian"
     And user "Brian" has created folder "userZeroFolder/userOneFolder"
-    When user "Brian" shares folder "userZeroFolder/userOneFolder" with user "Carol" with permissions "read, share" using the sharing API
-    And user "Carol" shares folder "userOneFolder" with user "Brian" using the sharing API
+    And user "Brian" has shared folder "userZeroFolder/userOneFolder" with user "Carol" with permissions "read, share"
+    When user "Carol" shares folder "userOneFolder" with user "Brian" using the sharing API
     Then the HTTP status code should be "200"
 #    Then the HTTP status code should be "405"
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiShareManagementBasicToRoot/createShareToRootFolder.feature:482]: @phil-davis do you have any context why this line is commented.
The scenario title has `sharing back to re-sharer is allowed`, but here, after sharing it back to the re-sharer, the HTTP code is a success but the folder is not there for re-sharer.

Is this a bug demonstration?

File: tests/acceptance/features/apiShareManagementBasicToRoot/createShareToRootFolder.feature [hunk 5]
```diff
@@ -473,8 +476,8 @@ Feature: sharing
     And user "Alice" has created folder "userZeroFolder"
     And user "Alice" has shared folder "userZeroFolder" with user "Brian"
     And user "Brian" has created folder "userZeroFolder/userOneFolder"
-    When user "Brian" shares folder "userZeroFolder/userOneFolder" with user "Carol" with permissions "read, share" using the sharing API
-    And user "Carol" shares folder "userOneFolder" with user "Brian" using the sharing API
+    And user "Brian" has shared folder "userZeroFolder/userOneFolder" with user "Carol" with permissions "read, share"
+    When user "Carol" shares folder "userOneFolder" with user "Brian" using the sharing API
     Then the HTTP status code should be "200"
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiShareManagementBasicToRoot/createShareToRootFolder.feature:481]: for sharing api, we can also assert ocs status codes. can be added in few more scenarios below too.
```
And the ocs status code should be ..
```

File: tests/acceptance/features/apiShareManagementBasicToRoot/createShareToRootFolder.feature [hunk 6]
```diff
@@ -504,11 +518,15 @@ Feature: sharing
     And user "Alice" has shared folder "userZeroFolder" with user "Brian"
     And user "Alice" has shared folder "userZeroFolder" with user "Carol"
     And user "Brian" has created folder "userZeroFolder/userOneFolder"
-    When user "Brian" shares folder "userZeroFolder/userOneFolder" with user "David" with permissions "read, share" using the sharing API
-    And user "David" shares folder "userOneFolder" with user "Carol" using the sharing API
-    Then the HTTP status code should be "200"
-#    Then the HTTP status code should be "405"
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiShareManagementBasicToRoot/createShareToRootFolder.feature:510]: what is this comment suggesting? should it be removed :thinking: ?

Comment (grgprarup) [reply tests/acceptance/features/apiShareManagementBasicToRoot/createShareToRootFolder.feature:510]: @kiranparajuli589 that commented line can be removed.

Comment (phil-davis) [reply tests/acceptance/features/apiShareManagementBasicToRoot/createShareToRootFolder.feature:510]: yes, the old suggestion of 405 "Method not allowed" was not correct. The systems-under-test give reasoable 404 "Not found" responses.

**Review by phil-davis — APPROVED** [review]

**Pull request merged**